resources/views/docs/1/digging-deeper/child-process.md

```diff
@@ -0,0 +1,300 @@
+on a regular schedule.
+
+Any scheduled tasks that would have run while your application isn't running will be skipped.
+
+**The queue runner and the scheduler are tied to your _application_, not the operating system, so they will
+only be able to run while your application is running.**
+
+### `shell_exec`, `proc_open` etc
+PHP has good built-in support for running arbitrary programs in separate processes. For example:
+
+- [`shell_exec`](https://www.php.net/manual/en/function.shell-exec.php) allows you to run commands and return their
+output to your application.
+- [`proc_open`](https://www.php.net/manual/en/function.proc-open.php) allows you to spin up a command with more control
+over how its input and output streams are handled.
+
+While these can be used in your NativePHP application, consider that they:
+- May block the script that is executing them until the sub-process has finished.
+- May become orphaned from your application, allowing them to continue running after your app has quit.
+
+Runaway orphaned processes could negatively impact your user's system and can become tricky to manage without user
+intervention. You should be cautious about starting processes this way.
+
+---
+
+## Starting a Child Process
+Each Child Process must have a unique alias. This is the name you will use to reference and interact with this process
+throughout your application.
+
+You may start a process using the `ChildProcess` facade:
+
+```php
+use Native\Laravel\Facades\ChildProcess;
+
+ChildProcess::start(
+    cmd: 'tail -n50 storage/logs/laravel.log',
+    alias: 'tail'
+);
+```
+
+The `start` method will return a `Native\Laravel\ChildProcess` instance, which represents the process. You may interact
+directly with this instance to make changes to that process, but this does not necessarily mean that the
+process was started.
+
+The timing of process initilization is controlled by the user's operating system and spawning
+may fail for a number of reasons.
+
+**To determine if the process has started successfully, you should listen for the
+[`ProcessSpawned` event](#codeprocessspawnedcode).**
+
+### Current Working Directory
+By default, the child process will use the working directory of your application as it's "current working directory"
+(`cwd`). However, you can explicitly change this if needed by passing a string path to the `$cwd` parameter of the
+`start` method:
+
+```php
+ChildProcess::start(
+    cmd: ['tail', '-n50', 'logs/laravel.log'],
+    alias: 'tail',
+    cwd: storage_path()
+);
+```
+
+### Persistent Processes
+You may mark a process as `persistent` to indicate that the runtime should make sure that once it has been started it
+is always running. This works similarly to tools like [`supervisord`](http://supervisord.org/), ensuring that the
+process gets booted up again in case it crashes.
+
+```php
+ChildProcess::start(
+    cmd: ['tail', '-n50', 'logs/laravel.log'],
+    alias: 'tail',
+    persistent: true
```

Comment:
same here if the point is to illustrate a long running process

otherwise this would restart every couple of milliseconds

Reply:
added in 3dc6cdae26e95b48610f6dd80bbcd881d5485412